### DISCUSSION TITLE
Error when assembling chunks, status code 504

### DISCUSSION THREAD

**Discussion by macaussie** [nextcloud/all-in-one#768] [Questions] [answered]
Hi. I was uploading a 3.6GB file via the web browser and received this error. On a page refresh the file does appear. Would this be to do with PHP max file size etc?

**Comment by szaimen** [comment]
How long did the upload take and can you post the complete log here?

**Reply by macaussie** [reply]
I never took note of the timing so I deleted and started again. It was 2 mins 30 secs for the upload then and 3 mins at processing files before the error appeared.

Hopefully this is the log you are looking for:

RAW

{"reqId":"lBvYvheMKWLDbfDGKgHh","level":3,"time":"2022-05-30T21:45:53+00:00","remoteAddr":"172.17.0.1","user":"admin","app":"PHP","method":"PUT","url":"/remote.php/dav/files/admin/._ubuntu-22.04-desktop-amd64.iso","message":"file_put_contents(/mnt/ncdata/admin/files/._ubuntu-22.04-desktop-amd64.iso): Failed to open stream: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#282","userAgent":"WebDAVFS/3.0.0 (03008000) Darwin/21.5.0 (x86_64)","version":"23.0.5.1","exception":{"Exception":"Error","Message":"file_put_contents(/mnt/ncdata/admin/files/._ubuntu-22.04-desktop-amd64.iso): Failed to open stream: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#282","Code":0,"Trace":[{"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::","args":[2,"file_put_contents(/mnt/ncdata/admin/files/._ubuntu-22.04-desktop-amd64.iso): Failed to open stream: Permission denied","/var/www/html/lib/private/Files/Storage/Local.php",282]},{"file":"/var/www/html/lib/private/Files/Storage/Local.php","line":282,"function":"file_put_contents","args":["/mnt/ncdata/admin/files/._ubuntu-22.04-desktop-amd64.iso",null]},{"file":"/var/www/html/lib/private/Files/Storage/Local.php","line":573,"function":"file_put_contents","class":"OC\\Files\\Storage\\Local","type":"->","args":["files/._ubuntu-22.04-desktop-amd64.iso",null]},{"file":"/var/www/html/lib/private/Files/Storage/Wrapper/Wrapper.php","line":647,"function":"writeStream","class":"OC\\Files\\Storage\\Local","type":"->","args":["files/._ubuntu-22.04-desktop-amd64.iso",null,"*** sensitive parameter replaced ***"]},{"file":"/var/www/html/custom_apps/files_antivirus/lib/AvirWrapper.php","line":94,"function":"writeStream","class":"OC\\Files\\Storage\\Wrapper\\Wrapper","type":"->","args":["files/._ubuntu-22.04-desktop-amd64.iso",null,"*** sensitive parameter replaced ***"]},{"file":"/var/www/html/lib/private/Files/Storage/Wrapper/Wrapper.php","line":647,"function":"writeStream","class":"OCA\\Files_Antivirus\\AvirWrapper","type":"->","args":["files/._ubuntu-22.04-desktop-amd64.iso",null,"*** sensitive parameter replaced ***"]},{"file":"/var/www/html/apps/dav/lib/Connector/Sabre/File.php","line":234,"function":"writeStream","class":"OC\\Files\\Storage\\Wrapper\\Wrapper","type":"->","args":["files/._ubuntu-22.04-desktop-amd64.iso",null]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php","line":1137,"function":"put","class":"OCA\\DAV\\Connector\\Sabre\\File","type":"->","args":[null]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":492,"function":"updateFile","class":"Sabre\\DAV\\Server","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPut","class":"Sabre\\DAV\\CorePlugin","type":"->","args":[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->","args":["method:PUT",[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->","args":[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/apps/dav/lib/Server.php","line":339,"function":"exec","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/remote.php","line":166,"args":["/var/www/html/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/html/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"id":"629541438a380"}


Formatted

[PHP] Error: file_put_contents(/mnt/ncdata/admin/files/._ubuntu-22.04-desktop-amd64.iso): Failed to open stream: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#282

PUT /remote.php/dav/files/admin/._ubuntu-22.04-desktop-amd64.iso
from 172.17.0.1 by admin at 2022-05-30T21:45:53+00:00

**Reply by szaimen** [reply]
5min is not a timeout that should happen with a pure AIO instance. Are you using it behind a reverse proxy? Also the error message seems to be unrelated..

**Reply by macaussie** [reply]
Yes it is behind Traefik2 using the rules I showed you before in the documentation.

**Reply by szaimen** [reply]
maybe we need to increase some timeouts of traefik?

**Comment by macaussie** [comment]
I shall run another vanilla instance of AIO without Traefik on a test domain and see what happens.

**Reply by szaimen** [reply]
Thanks for testing! :)

**Reply by szaimen** [reply]
I am also trying again on my test instance. Since it seems to happen for you after a few minutes I suppose it should be easily reproducible  on a vanilla instance...

**Reply by macaussie** [reply]
Yeah that's what I am thinking. Containers just starting now.

**Reply by szaimen** [reply]
just tested and works for me without any problem...

(tested with a 1,4GB file which took around 5min....)

**Comment by szaimen** [comment]
the max file size is btw set to 10GB and max execution time to 1h if I remember correctly...

**Reply by macaussie** [reply]
Yes worked for me on the test domain. Must be something to do with the reverse proxy (Traefik2). I shall work on that later. 

Thank you.

**Reply by szaimen** [reply]
Thanks!

**Reply by szaimen** [reply]
any update here @macaussie ? :)

**Reply by macaussie** [reply]
I'm in the process of moving house. Everything is in storage at the moment.

**Reply by szaimen** [reply]
No problem, there's no hurry!

Good luck with moving!

**Comment by szaimen** [comment, marked as answer]
Seems like this may help a bit with this problem: https://github.com/nextcloud/all-in-one/pull/1178

**Reply by szaimen** [reply]
This is now released with v2.0.3 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
